components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageTxImpl.java

```diff
@@ -200,12 +215,19 @@ private OrientVertex findWithPropertyOwnedBy(String className, String propName,
 
   @Override
   @Guarded(by = OPEN)
-  public Iterable<OrientVertex> findAssets(@Nullable String whereClause,
-                                           @Nullable Map<String, Object> parameters,
-                                           @Nullable Iterable<Repository> repositories,
-                                           @Nullable String querySuffix)
+  public Iterable<Asset> findAssets(@Nullable String whereClause,
+                                    @Nullable Map<String, Object> parameters,
+                                    @Nullable Iterable<Repository> repositories,
+                                    @Nullable String querySuffix)
   {
-    return findVertices(V_ASSET, whereClause, parameters, E_OWNS_ASSET, repositories, querySuffix);
+    return Iterables.transform(
+        findVertices(V_ASSET, whereClause, parameters, E_OWNS_ASSET, repositories, querySuffix),
+        new Function<OrientVertex, Asset>() {
```

Comment:
might want to extract the functions used to constants, this one is a repeat but we also shouldn't need to instantiate these more than once